### DISCUSSION TITLE
Adding some prompt templates into the front page of Web App

### DISCUSSION THREAD

**Discussion by itkrmstk** [microsoft/sample-app-aoai-chatGPT#215] [Ideas]
Hello, 
I'm customizing this sample code for the development of an in-house Chat Web App. I'm thinking of placing several prompt templates on the frontpage to ensure that in-house users don't get confused about how to use it. Please see an image for reference.
<img width="486" alt="Prompt templates" src="https://github.com/microsoft/sample-app-aoai-chatGPT/assets/63496455/a88d94c5-7ac6-472f-859f-2305a3b50397">

As a simple method, I'm considering having pre-set text input into the message box (user input columns which under the prompt templates) when a prompt template is clicked.

For more advanced prompt templates, is it possible to change the system message itself based on the prompt template?

If you have any sample code or resources that have already implemented the above, I would greatly appreciate it if you could share them.

Regards,

**Comment by sarah-widder** [comment]
Hi @itkrmstk you might check out this repo, which provides very similar code for a chat app: https://github.com/Azure-Samples/azure-search-openai-demo/tree/main but it doesn't use Azure OpenAI On Your Data.

Specifically, the ExampleList here https://github.com/Azure-Samples/azure-search-openai-demo/blob/62e22486632194ca387ba3c7187c4db4db8a5425/app/frontend/src/pages/chat/Chat.tsx#L176C29-L176C80 may be what you're looking for.

As for changing the system message based on the prompt template, you can customize the backend flow that calls Azure OpenAI however you'd like. Perhaps pass some additional info from the frontend to the backend when an example prompt is picked, and use that to map to a specific system message in the backend before calling chat completions.

**Reply by itkrmstk** [reply]
Hello @sarah-widder , and thanks for your reply!
I found the relative "Example" components in your  refference post and use in my own code.
Thanks and I would close this post if there is no additional idea or discussion.

**Reply by ghost** [reply]
@itkrmstk Did you manage to make it work? If yes, can you please elaborate on that. I'm also looking for a similar option

**Comment by jeannot-prvlimburg** [comment]
Any updates on this?
